packages/arb-avm-cpp/avm/src/machinestate/machineoperation.cpp

```diff
@@ -971,8 +974,10 @@ BlockReason sideload(MachineState& m) {
         m.context.sideloads.pop_back();
     } else {
         if (m.context.stop_on_sideload && !m.context.first_instruction) {
+            m.l2_block_number = block_num;
             return SideloadBlocked{block_num};
         }
+        m.last_inbox_timestamp = 0;
```

Comment:
I don't see why we reset the timestamp here

Reply:
I was thinking of resetting the values when there isn't any sideloads, but I don't think it makes much sense in retrospect, will remove.